src/comparator/src/index.ts

```diff
@@ -411,7 +411,9 @@ export default class BuildComparator {
     const groups = this.getStringFormattedGroups(formatTotal, formatDelta, sizeKey);
     const rows = this.getStringFormattedRows(formatTotal, formatDelta, sizeKey, artifactFilter);
 
-    return markdownTable([header, ...groups, ...rows], { align: rows[0].map((_, i) => (i === 0 ? 'l' : 'r')) });
```

Comment:
There are a number of other places where we're assuming the array will always be non-empty (i.e. `xs[0].map(…)`), which might suffer from similar bugs if it turns out the array might in fact be empty.

Reply:
A build shouldn't be allowed in by the API that has less than one artifact. Is there another case where this is possible?

Reply:
Not possible to use optional chaining here?

Reply:
> Is there another case where this is possible?

I believe so—see the test I added. `artifactFilter` may return false for all items. This is actually what happened to me when testing the example used here: https://buildtracker.dev/docs/guides/ci

> Not possible to use optional chaining here?

Sure, I'll give it a shot!

Reply:
Ah filters, of course!

I didn't do anything to enable optional chaining—not sure if it's enabled by default. Happy to start including it, though!

Reply:
I'm not sure the optional chaining operator can help here, but I tried to use nullish coalescing. Unfortunately it seems the version of ESLint that is currently used does not have support:

![image](https://user-images.githubusercontent.com/921609/74669342-dc3eb080-519e-11ea-90ed-9071d644017c.png)

```sh-session
❯ npm run lint

> build-tracker@1.0.0 lint /Users/oliverash/Development/build-tracker
> yarn lint:cmd --fix

yarn run v1.19.1
$ eslint . --ext '.ts,.tsx,.js,.json,.md,.snap' --cache --fix
{ parser: "babylon" } is deprecated; we now treat it as { parser: "babel" }.

/Users/oliverash/Development/build-tracker/src/comparator/src/index.ts
  414:30  error  Parsing error: Expression expected
```

Reply:
I plan on upgrading eslint, typescript, and a few other things today